libfreerdp/utils/smartcard_call.c

```diff
@@ -43,23 +43,96 @@
 #include <freerdp/log.h>
 #define TAG FREERDP_TAG("utils.smartcard.call")
 
-#if defined(WITH_SMARTCARD_EMULATE)
+#ifdef WITH_SMARTCARD_EMULATE
 #include <freerdp/emulate/scard/smartcard_emulate.h>
 
-#define str(x) #x
-#define wrap(ctx, fkt, ...) Emulate_##fkt(ctx->emulation, ##__VA_ARGS__)
-#else
-#define wrap(ctx, fkt, ...) fkt(__VA_ARGS__)
+#define WRAP_WINSCARD_CALL_RET(ctx, fkt, _ret, ...)              \
```

Comment:
Why make the initial macro this complicated? `#define wrap(ctx, fkt, ...) ctx->emulated ? Emulate_##fkt(ctx->emulation, ##__VA_ARGS__) : fkt(__VA_ARGS__)` also works and does not need that many changes to original code ;)